lib/Backends/NNPI/Importer.cpp

```diff
@@ -0,0 +1,1645 @@
+                   m_writeTensors.count(v->getName()));
+    if (inputVar || outputVar) {
+      LOG_NNPI_ERROR_RETURN_INVALID_HANDLE(
+          AddValue(v->getName(), v->getType(),
+                   IsVariableUsingAlternativeLayout(v), inputVar, outputVar),
+          "Failed to add placeholder");
+      // DBG("[--IO--] Setting IO variable: " << v->getName() << ", R:" <<
+      // inputVar << ", W:" << outputVar << ", U:" << v->getNumUsers());
+    } else {
+      // DBG("[--IO--] Unused Placeholder: " << v->getName());
+    }
+  }
+
+  DBG_MEM_USAGE("ImportFunction call nnpiNetworkBuild");
+  // Build network
+  NNPINetwork net;
+  NNPIErrorCode res = nnpiNetworkBuild(m_network);
+  if (res != NNPI_NO_ERROR) {
+    // todo handle errors better
+    LOG(INFO) << "Failed to build network";
+    LOG_NNPI_ERROR(nnpiNetworkDestroy(m_network),
+                   "Failed to destroy NNPI network");
+    net = NNPI_INVALID_NNPIHANDLE;
+  } else {
+    net = m_network;
+  }
+
+  // detach network from importer (if failed to build then it's already
+  // destroyed, otherwise relinquish ownership to the backend)
+  m_network = NNPI_INVALID_NNPIHANDLE;
+  DBG_MEM_USAGE("ImportFunction done >>");
+  return net;
+}
+
+// Node Importers ////////////////////////////////////////////////////////
+class ConvolutionNodeImporter : public INNPINodeImporter {
+public:
+  NNPIErrorCode ImportNode(Node *n, NNPIImporter &importer) override {
+    auto *glowConv = llvm::dyn_cast<ConvolutionNode>(n);
+    LOG_AND_RETURN_IF_NOT(ERROR, glowConv, "Bad node type", NNPI_INVALID_PARAM);
+
+    const uint32_t SPATIAL_DIMS2 = 2;
+    LOG_AND_RETURN_IF_NOT(ERROR, glowConv->getKernels().size() == SPATIAL_DIMS2,
+                          "[Conv] Invalid number of kernel sizes",
+                          NNPI_INVALID_PARAM);
+    LOG_AND_RETURN_IF_NOT(ERROR,
+                          glowConv->getPads().size() == 2 * SPATIAL_DIMS2,
+                          "[Conv] Invalid number of pads", NNPI_INVALID_PARAM);
+    LOG_AND_RETURN_IF_NOT(ERROR, glowConv->getStrides().size() == SPATIAL_DIMS2,
+                          "[Conv] Invalid number of strides",
+                          NNPI_INVALID_PARAM);
+
+    uint32_t kernel[SPATIAL_DIMS2] = {glowConv->getKernels()[0],
+                                      glowConv->getKernels()[1]};
+    uint32_t paddingStart[SPATIAL_DIMS2] = {glowConv->getPads()[0],
+                                            glowConv->getPads()[1]};
+    uint32_t paddingEnd[SPATIAL_DIMS2] = {glowConv->getPads()[2],
+                                          glowConv->getPads()[3]};
+    uint32_t stride[SPATIAL_DIMS2] = {glowConv->getStrides()[0],
+                                      glowConv->getStrides()[1]};
+    uint32_t dilation[SPATIAL_DIMS2] = {glowConv->getDilation(),
+                                        glowConv->getDilation()};
+
+    LOG_NNPI_ERROR_RETURN_VALUE(
+        importer.AddTensor(NodeValueName(glowConv->getFilter()), true),
```

Comment:
nit: `/* alternativeLayout */ true` etc. everywhere when passing raw booleans to functions.

Reply:
Done